src/cpp/session/projects/SessionProjectContext.cpp

```diff
@@ -183,23 +183,21 @@ Error computeScratchPaths(const FilePath& projectFile,
                           FilePath* pScratchPath,
                           FilePath* pSharedScratchPath)
 {
-   // ensure project user dir
+   // compute project user dir
    FilePath projectUserDir = computeUserDir(projectFile, projectConfig);
-   if (!projectUserDir.exists())
-   {
-      // create
-      Error error = projectUserDir.ensureDirectory();
-      if (error)
-         return error;
+   
+   // make sure it exists
+   Error error = projectUserDir.ensureDirectory();
```

Comment:
`ensureDirectory()` already calls `exists()` internally, so it's unnecessary to call it here.